app/src/main/java/unipd/se18/ocrcamera/MignifyResolver.java

```diff
@@ -0,0 +1,111 @@
+        try {
+            response = Unirest
+                    .get(fullURL)
+                    //Authentication header
+                    .header("X-Mashape-Key",
+                            MASHAPE_AUTH)
+                    //Accept and Content-Type selection header
+                    .header("Accept", "application/json")
+                    //Type of response we expect
+                    .asJson();
+        } catch (UnirestException e) {
+            Log.e("Unirest error", e.getMessage());
+        }
+        if (response == null) {
+            product = HTTP_ERROR;
+        } else {
+            try {
+                //Retrieve the parsed JSONObject from the Mashape response
+                JSONObject jsonResponse = response.getBody().getObject();
+                //Getting the status from the JSON
+                //The server responds with either "ok" or "error"
+                String status = jsonResponse.getString("status");
+                if (status.equals("ok")) {
+                    //If the EAN code was found, in the JSON there is an list of products, each
+                    //one with name, brand and language
+                    JSONArray productList = jsonResponse.getJSONArray("results");
+                    String[] productNames = new String[productList.length()];
+                    //Future proofing usage of brands and language, still unimplemented
+                    String[] productBrands = new String[productList.length()];
+                    String[] productLanguages = new String[productList.length()];
+                    //If no product are in the response
+                    if (jsonResponse.length() == 0) {
+                        product = NO_PRODUCT_ERROR;
+                    } else {
+                        //Getting name, brand and language for each product
+                        for (int i = 0; i < jsonResponse.length(); i++) {
+                            //Getting the current product from the JSON list and it's values
+                            JSONObject currentJsonObject = productList.getJSONObject(i);
+                            productNames[i] = currentJsonObject.getString("productName");
+                            productBrands[i] = currentJsonObject.getString("brand");
+                            productLanguages[i] = currentJsonObject.getString("languageCode");
+                        }
+
+                        //Selecting the first name in the list as the result product name.
+                        //It should be the most plausible product name.
+                        //In the future a better selection will be made, checking brand and language
+                        product = productNames[0];
+                    }
+                } else if(status.equals("error")){
+                    //In case the EAN wasn't valid, the error in the JSON will be returned
+                    product = jsonResponse.getString("message");
```

Comment:
Are you sure this error message is useful for the user who is looking for the product corresponding to the barcode? Maybe you could assign to product variable a more meaningful message to explain the user what has happened.

Reply:
I probably should have explained this part of the code better.
Basically, this is the case where the barcode is invalid, which will never happens since the Barcode Module already check for that.
I checked anyway what the API would respond to a wrong barcode, and it was a good and understandable response.
I do agree I should normalize the response with an error string I decide, but if wasn't urging since this else should never be entered.